src/main/kotlin/com/tylerthrailkill/helpers/prettyprint/PrettyPrint.kt

```diff
@@ -1,255 +1,184 @@
 package com.tylerthrailkill.helpers.prettyprint
 
 import com.ibm.icu.text.BreakIterator
+
 import mu.KotlinLogging
-import java.io.PrintStream
-import java.util.*
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.util.UUID
 
-typealias CurrentNodeIdentityHashCodes = ArrayDeque<Int>
-typealias DetectedCycles = MutableList<Int>
+const val DEFAULT_WRAP_WIDTH = 80
+const val DEFAULT_INDENT = 2
+val DEFAULT_OUT: Appendable = System.out
 
 /**
  * Pretty print function.
  *
  * Prints any object in a pretty format for easy debugging/reading
  *
- * @param[obj] the object to pretty print
- * @param[tabSize] optional param that specifies the number of spaces to use to indent
- * @param[printStream] optional param that specifies the [PrintStream] to use to pretty print. Defaults to `System.out`
+ * @param [obj] the object to pretty print
+ * @param [indent] optional param that specifies the number of spaces to use to indent. Defaults to 2.
+ * @param [writeTo] optional param that specifies the [Appendable] to output the pretty print to. Defaults appending to `System.out`.
+ * @param [wrappedLineWidth] optional param that specifies how many characters of a string should be on a line.
  */
-fun pp(obj: Any?, tabSize: Int = 2, printStream: PrintStream = System.out, wrappedLineWidth: Int = 80) {
-    val prettyPrint = PrettyPrint(tabSize, printStream, wrappedLineWidth)
-    val nodeList = CurrentNodeIdentityHashCodes()
-    val detectedCycles: DetectedCycles = mutableListOf()
-    prettyPrint.recurse(obj, nodeList, detectedCycles, "")
-    printStream.println()
-}
+fun pp(
+    obj: Any?,
+    indent: Int = DEFAULT_INDENT,
+    writeTo: Appendable = DEFAULT_OUT,
+    wrappedLineWidth: Int = DEFAULT_WRAP_WIDTH
```

Comment:
I hadn't thought of looking at the IDE's hints. I think that since these constants are not used anywhere else and are very specific to this function, we can hardcode them into the function's signature (so I'm favoring the former)